### PR TITLE
Move add-row buttons into tfoot

### PR DESCRIPTION
**Pull request by mccalluc**
It will simplify the JS if we have more generic event handlers: Buttons can add rows to the tables where they are located, instead of needing one handler per button.

This PR moves the buttons, and consolidates the JS for row-reordering.

- Pattern demonstrated in #759
- Towards #761

### CODE REVIEW

File: app/views/works/_related_objects_table.html.erb [hunk 1]
```diff
@@ -4,14 +4,22 @@
...
         <th>Related Identifier Type</th>
         <th>Relation Type</th>
       </tr>
-    </tbody>
+    </thead>
+    <tbody id="related-objects-table"></tbody>
+    <tfoot>
+      <tr>
+        <td colspan="3">
+          <%= link_to "Add Another Related Object", "#", id: "btn-add-related-object", class: "btn btn-secondary" %>
```
Comment (carolyncole) [inline app/views/works/_related_objects_table.html.erb:19]: Is there code that should be deleted from Related?  Maybe this needs a rebase??

Comment (mccalluc) [reply app/views/works/_related_objects_table.html.erb:19]: I'm not sure that I understand: The button that's added just above was moved/deleted from farther down in the file.

Comment (carolyncole) [reply app/views/works/_related_objects_table.html.erb:19]: The PR seems to indicate that this is a refactor.  This seems to add new functionality.

Comment (mccalluc) [reply app/views/works/_related_objects_table.html.erb:19]: I don't understand your comment: If we're talking about this line, the `Add Another Related Object` is being moved from a div underneath the table to a tfoot inside the table (which corresponds to the title on the PR).

Let me know what I'm missing / what needs to change?

**Review by carolyncole — APPROVED** [review]

**Pull request merged**